homeassistant/components/blebox/climate.py

```diff
@@ -40,23 +53,31 @@ class BleBoxClimateEntity(BleBoxEntity[blebox_uniapi.climate.Climate], ClimateEn
     """Representation of a BleBox climate feature (saunaBox)."""
 
     _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
-    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT]
     _attr_temperature_unit = TEMP_CELSIUS
 
+    @property
+    def hvac_modes(self):
+        """Return list of supported HVAC modes."""
+        return [HVACMode.OFF, self.hvac_mode]
+
     @property
     def hvac_mode(self):
         """Return the desired HVAC mode."""
         if self._feature.is_on is None:
             return None
```

Comment:
When does this happen? The code below checks for a boolean on this property.

Reply:
This is unlikely to happen. Although blebox_uniapi might return `None `.